TypeCobol.LanguageServer/VsCodeProtocol/LanguageServer.cs

```diff
@@ -41,15 +41,42 @@ public LanguageServer(IRPCServer rpcServer)
 
             RemoteConsole = new RemoteConsole(rpcServer);
             RemoteWindow = new RemoteWindow(rpcServer);
+
+            //Track any unhandled exception during rpc communication from the server
+            rpcServer.Handler += UnhandledExceptionHandler;
+            //Also enforce with AppDomain unhandled exception handler
+            AppDomain currentDomain = AppDomain.CurrentDomain;
+            currentDomain.UnhandledException += UnhandledExceptionHandler;
+        }
+
+        /// <summary>
+        /// Destructor.
+        /// </summary>
+        ~LanguageServer()
+        {
+            this.RpcServer.Handler -= UnhandledExceptionHandler;
+            AppDomain currentDomain = AppDomain.CurrentDomain;
+            currentDomain.UnhandledException -= UnhandledExceptionHandler;
```

Comment:
I don't think we need to use the `AppDomain.UnhandledException` event if we aready have our own logic for handling exceptions in the server object but we do have to ensure that this last chance handling is always safe so add a try-catch block around the `NotifyException` call in the handler.
Also why attempt to unregister the handler in a finalizer, the application is shutting down anyway.

Reply:
AppDomain.UnhandledException is interresting to capture all exceptions that we don't know yet.

Reply:
The destructor here to ensure safe programming.

Reply:
Ok I think I see now, we have 2 kinds of scenarios: some exceptions are already handled but not properly notified (those are the ones in the various catch block of JsonRPCServer and we also have unpredictable exceptions (those can be handled with the AppDomain.UnhandledException event as a last resort. 

It would be best to remove the Handler property from IRPCServer interface and add a new parameter in JsonRPCServer constructor but it cannot currently easily be done. So ok to keep the property.

As for the destructor, I maintain my previous remark. There is no need to detach handling methods because in our current architecture, if LanguageServer dies, it means the whole process is exiting.